aviary/interface/reports.py

```diff
@@ -172,3 +180,95 @@ def _get_phase_diff(traj, phase, var_name, units, indices=[0, -1]):
+        includes='*timeseries*', out_stream=None, return_format='dict', units=True)
+    phase_names = prob.model.traj._phases.keys()
+
+    timeseries_outputs = {value['prom_name']: value for key,
+                          value in timeseries_outputs.items()}
+
+    timeseries_outputs = {key: value for key,
+                          value in timeseries_outputs.items() if not key.endswith('_phase')}
+
+    unique_variable_names = set([timeseries_output.split('.')[-1]
+                                for timeseries_output in timeseries_outputs])
+
+    timeseries_data = {}
+    for variable_name in unique_variable_names:
+        timeseries_data[variable_name] = {}
+        val_full_traj = np.zeros((0, 1))
+        units = None
+        for idx_phase, phase_name in enumerate(phase_names):
+            variable_str = f'traj.{phase_name}.timeseries.{variable_name}'
+            time_str = f'traj.{phase_name}.timeseries.time'
+
+            if variable_str not in timeseries_outputs:
+                Warning(
+                    f'Variable {variable_str} not found in timeseries_outputs for phase {phase_name}.')
+                val = np.zeros_like(timeseries_outputs[time_str]['val'])
+                val[:] = np.nan
+                val_full_traj = np.vstack((val_full_traj, val))
+
+            else:
+                val = timeseries_outputs[variable_str]['val']
+
+                # grab the units from the first phase that uses this variable; use these units for all others
+                if units is None:
+                    units = timeseries_outputs[variable_str]['units']
+                    val_full_traj = np.vstack((val_full_traj, val))
+                else:
+                    original_units = timeseries_outputs[variable_str]['units']
+
+                    if original_units != units:
+                        val = wrapped_convert_units((val, original_units), units)
+
+                    val_full_traj = np.vstack((val_full_traj, val))
+
+        timeseries_data[variable_name]['val'] = val_full_traj
+        timeseries_data[variable_name]['units'] = units
+        timeseries_data[variable_name]['shape'] = val_full_traj.shape
+
+    # Create a DataFrame from timeseries_data
+    df_data = {variable_name: pd.Series(timeseries_data[variable_name]['val'].flatten())
+               for variable_name in timeseries_data}
+    df = pd.DataFrame(df_data)
+
+    time_column = ['time']  # Isolate the 'time' column
+    # Sort the rest of the columns
+    other_columns = sorted([col for col in df.columns if col != 'time'])
+    columns = time_column + other_columns  # Combine them, keeping 'time' first
+    df = df[columns]
+
+    # Add units to column names
+    df.columns = [f'{col} ({timeseries_data[col]["units"]})' for col in df.columns]
+
+    df.drop_duplicates()
+
+    # The path where you want to save the CSV file
+    reports_folder = Path(prob.get_reports_dir())
+    report_file = reports_folder / 'mission_timeseries_data.csv'
+
```

Comment:
In MPI, we probably only want to print the file on rank 1, so insert these lines right before we start printing.

``
    if MPI and MPI.COMM_WORLD.rank != 0:
        return
```

Reply:
Great call, thanks for that addition!